app/services/PaymentService.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.Period
+import models.requests.{PaymentPeriod, PaymentRequest}
+import uk.gov.hmrc.domain.Vrn
+
+import javax.inject.Inject
+
+class PaymentService @Inject()() {
+
+  def buildPaymentRequest(vrn: Vrn, period: Period, amount: Long): PaymentRequest = {
```

Comment:
Should amount here be a BigDecimal?  At this point I think it's the amount in pounds, so treating it as a long will round the value